### PR TITLE
helpers: escape name

### PR DESCRIPTION
**Pull request by greut**
**What this PR does / why we need it**:

1. I've created a helm chart named `no`
2. things blew up, because it's not a string literal.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: pkg/chartutil/create.go [hunk 1]
```diff
@@ -435,7 +435,7 @@ app.kubernetes.io/managed-by: {{ .Release.Service }}
 Selector labels
 */}}
 {{- define "<CHARTNAME>.selectorLabels" -}}
-app.kubernetes.io/name: {{ include "<CHARTNAME>.name" . }}
+app.kubernetes.io/name: {{ include "<CHARTNAME>.name" . | toJson }}
```
Comment (mattfarina) [inline pkg/chartutil/create.go:438]: Why is the name being piped to `toJson`?

Comment (greut) [reply pkg/chartutil/create.go:438]: The `name: %q` above fixes this.

```console
$ helm create no                                                                                                                                                                                     
Creating no

$ helm template ./no | grep 'false'
# Source: false/templates/serviceaccount.yaml
  name: RELEASE-NAME-false
    helm.sh/chart: false-0.1.0
    app.kubernetes.io/name: false
# Source: false/templates/service.yaml
  name: RELEASE-NAME-false
    helm.sh/chart: false-0.1.0
    app.kubernetes.io/name: false
    app.kubernetes.io/name: false
# Source: false/templates/deployment.yaml
  name: RELEASE-NAME-false
    helm.sh/chart: false-0.1.0
    app.kubernetes.io/name: false
      app.kubernetes.io/name: false
        app.kubernetes.io/name: false
      serviceAccountName: RELEASE-NAME-false
        - name: false
# Source: false/templates/tests/test-connection.yaml
  name: "RELEASE-NAME-false-test-connection"
    helm.sh/chart: false-0.1.0
    app.kubernetes.io/name: false
      args: ['RELEASE-NAME-false:80']
```

This one, fixes that.

```
helm template ./no | grep -P '\bno\b'
# Source: no/templates/serviceaccount.yaml
  name: RELEASE-NAME-no
    helm.sh/chart: no-0.1.0
    app.kubernetes.io/name: no
# Source: no/templates/service.yaml
  name: RELEASE-NAME-no
    helm.sh/chart: no-0.1.0
    app.kubernetes.io/name: no
    app.kubernetes.io/name: no
# Source: no/templates/deployment.yaml
  name: RELEASE-NAME-no
    helm.sh/chart: no-0.1.0
    app.kubernetes.io/name: no
      app.kubernetes.io/name: no
        app.kubernetes.io/name: no
      serviceAccountName: RELEASE-NAME-no
        - name: no
# Source: no/templates/tests/test-connection.yaml
  name: "RELEASE-NAME-no-test-connection"
    helm.sh/chart: no-0.1.0
    app.kubernetes.io/name: no
      args: ['RELEASE-NAME-no:80']
```

Comment (technosophos) [reply pkg/chartutil/create.go:438]: I think that instead of `| toJson`, you want `| quote`

**Comment by technosophos** [comment]
This needs an issue # and a test. PRs must reference an issue. I left a note about using `| quote` instead of `| toJson`.

**Comment by bacongobbler** [comment]
closing as inactive. Multiple maintainers have chimed in here and no response has been provided.

When you are ready for another review, please open a new PR and reference this ticket. Thanks.

**Pull request closed without merging**